algorithms/longest non-repeat substring/Solution.java

```diff
@@ -0,0 +1,28 @@
+/**
```

Comment:
Is this file naming not the same with the rest?
like "longest-substring-without-repeating-characters"
I would suggest we do exact matching for url